### DISCUSSION TITLE
Update docs for slice pattern and usage of middleware such as devtools example included

### DISCUSSION THREAD

**Discussion by enyelsequeira** [pmndrs/zustand#1776] [Ideas]
##  I was talking with @dai-shi about this and maybe this should be an issue to update

ideally it would be great to have a bit better documentation about the different slices patterns and how they can work or be used in different stores, example like this below

```js 
const useStore = create<
  UserSlice & UserActionSlice & TokensSlice & PasscodeAndEmailSetupSlice
>()((...a) => ({
  ...CreateUserSlice(...a),
  ...UserActions(...a),
  ...PassCodeAndEmail(...a),
  ...TokenValues(...a),
}));
```

How can I update the store from another one if I am using the slices patterns
and yes, we have an example of how they work, but I think maybe there should be an depth documentation to better help other developers

Cheers,

**Comment by dai-shi** [comment]
@chrisk-7777 any thoughts on this?

**Comment by chrisk-7777** [comment]
100% agree. I feel like we also had this in the wiki at some point. 

I've added the documentation label and believe it should stay open.

**Comment by dai-shi** [comment]
@dbritto-dev Would you be interested in this?

**Reply by dbritto-dev** [reply]
@dai-shi yeah, I would like to work on that
